### PR TITLE
2 task components

### PR DESCRIPTION
**Pull request by otecpavlo**
1. Task: https://github.com/rolling-scopes-school/tasks/blob/master/tasks/angular/components-directives-pipes.md
2. Screenshot:
   Imgur
3. Deploy: https://otecpavlo.github.io/youtube-client/youtube-client/bundle/
4. Done10.08.2021 / deadline 08.08.2021
:heavy_check_mark: Main layout is implemented
:heavy_check_mark: Implemented a fake search functionality: by submitting the form on the Header component, search results are shown
:heavy_check_mark: The Search result item component contains all the necessary data 
:heavy_check_mark: The Filtering criteria block toggle functionality is implemented
:heavy_check_mark: The colorful border under the Search result item is implemented 
:heavy_check_mark: Search result sorting is implemented (Should work both in the direction of decreasing values, and in the direction of increasing values) 
:heavy_check_mark: Search result filtering by key words is implemented

### CODE REVIEW

File: youtube-client/src/app/services/search-service.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  public searchItems: ISearchItem[] = [];
+
+  constructor(private http: HttpClient) { }
+
+  searchRequest(): void {
+    const URL = 'https://raw.githubusercontent.com/rolling-scopes-school/tasks/master/tasks/angular/response.json';
+    this.http.get<ISearchResponse>(URL)
+      .subscribe((response) => {
+        this.searchItems.length = 0;
+        response.items.forEach((item) => {
```
Comment (4m9fk) [inline youtube-client/src/app/services/search-service.ts:19]: You should use `map` for it.
```ts
this.searchItems = response.items.map( item => ({...}))
```

File: youtube-client/src/app/services/search-service.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      this.searchItems.sort((a, b) => {
+        if ((new Date(a.publishedDate)) > (new Date(b.publishedDate))) {
+          return 1;
+        }
+        return -1;
+      });
+    }
+  }
+
+  sortByViews(directionToBottom: boolean) {
```
Comment (4m9fk) [inline youtube-client/src/app/services/search-service.ts:54]: Let's use pipe for sorting and filtering.
You can check example here https://github.com/ppparfenovich/youtube-client/pull/2/files#diff-f5dbdfa92848dde3da3141f676253d75d351cdb029db006226240c8ac8dd285c

File: youtube-client/src/app/search-result/directives/colorful-border.directive.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import {
+  Directive, ElementRef, Input, OnInit, Renderer2
+} from '@angular/core';
+
+@Directive({
+  selector: '[colorfulBorder]'
+})
+export class ColorfulBorderDirective implements OnInit {
```
Comment (4m9fk) [inline youtube-client/src/app/search-result/directives/colorful-border.directive.ts:8]: Great that you used directive!

File: youtube-client/src/app/search-result/search-result-item/search-result-item.component.scss [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+// .comments-icon {
+//   display: inline-block;
+//   width: 18px;
+//   height: 18px;
+//   background: center / cover no-repeat url('../../../assets/comments.png');
+//   margin: 0 3px;
+// }
+// .comments-icon__span {
```
Comment (4m9fk) [inline youtube-client/src/app/search-result/search-result-item/search-result-item.component.scss:61]: PLease remove commented code

File: youtube-client/src/app/search-result/search-result-item/search-result-item.component.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { ISearchItem } from '../../models/search-result-item.model';
+
+@Component({
+  selector: 'search-result-item',
+  templateUrl: './search-result-item.component.html',
+  styleUrls: ['./search-result-item.component.scss']
+})
+export class SearchResultItemComponent implements OnInit {
+  @Input() item!: ISearchItem;
```
Comment (4m9fk) [inline youtube-client/src/app/search-result/search-result-item/search-result-item.component.ts:10]: Please avoid `!` usage

**Review by 4m9fk — CHANGES_REQUESTED** [review]

File: youtube-client/src/app/search-result/search-result-item/search-result-item.component.scss (hunk 4 above)
Comment (otecpavlo) [reply youtube-client/src/app/search-result/search-result-item/search-result-item.component.scss]: removed

File: youtube-client/src/app/search-result/search-result-item/search-result-item.component.ts (hunk 5 above)
Comment (otecpavlo) [reply youtube-client/src/app/search-result/search-result-item/search-result-item.component.ts]: `!` removed

File: youtube-client/src/app/services/search-service.ts (hunk 2 above)
Comment (otecpavlo) [reply youtube-client/src/app/services/search-service.ts]: added sort and filter pipes

File: youtube-client/bundle/3rdpartylicenses.txt [hunk 6]
```diff
@@ -0,0 +1,241 @@
+@angular/common
```
[the remark this answers is not in the document]
Comment (otecpavlo) [reply youtube-client/bundle/3rdpartylicenses.txt]: fixed

File: youtube-client/src/app/services/search-service.ts (hunk 1 above)
Comment (otecpavlo) [reply youtube-client/src/app/services/search-service.ts]: fixed

**Pull request closed without merging**